idaes/apps/flexibility_analysis/decision_rules/relu_dr.py

```diff
@@ -0,0 +1,160 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+This module contains a function for constructing a neural network-based decision 
+rule for the inner problem of the flexibility test.
+"""
+from typing import MutableMapping, Sequence
+import numpy as np
+from pyomo.common.dependencies import attempt_import
+import pyomo.environ as pe
+from pyomo.core.base.var import _GeneralVarData
+from pyomo.core.base.block import _BlockData
+from idaes.apps.flexibility_analysis.indices import _VarIndex
+from .relu_dr_config import ReluDRConfig
+
+tf, tensorflow_available = attempt_import("tensorflow")
+keras, keras_available = attempt_import("tensorflow.keras")
+omlt, omlt_available = attempt_import("omlt")
+omlt_nn, _ = attempt_import("omlt.neuralnet")
+omlt_io, _ = attempt_import("omlt.io")
+plt, _ = attempt_import("matplotlib.pyplot")
+
+
+def construct_relu_decision_rule(
+    input_vals: MutableMapping[_GeneralVarData, Sequence[float]],
+    output_vals: MutableMapping[_GeneralVarData, Sequence[float]],
+    config: ReluDRConfig,
+) -> _BlockData:
+    """
+    Construct a neural network-based decision rule with ReLU activation functions
+    from the data provided for the inputs and outputs.
+
+    Parameters
+    ----------
+    input_vals: input_vals: MutableMapping[_GeneralVarData, Sequence[float]]
+        Data for the variables that are inputs to the decision rule
+    output_vals: input_vals: MutableMapping[_GeneralVarData, Sequence[float]]
+        Data for the variables that are outputs to the decision rule
+    config: ReluDRConfig
+        A config object to specify options for the decision rule
+
+    Returns
+    -------
+    res: _BlockData
+        A pyomo model containing the linear decision rule
+    """
+    tf.random.set_seed(config.tensorflow_seed)
+    inputs = list(input_vals.keys())
+    outputs = list(output_vals.keys())
+    n_samples = len(input_vals[inputs[0]])
+
+    config: ReluDRConfig = config()
+    if config.batch_size > n_samples:
+        config.batch_size = n_samples
```

Comment:
It might be good to log a message for the user here to let them know they set a batch size greater than the number of samples and thus this value was changed.